dali/kernels/imgproc/convolution/convolution_gpu_test.cu

```diff
@@ -0,0 +1,193 @@
+      int window_size = shape_window[sample][0];
+      for (int i = 0; i < window_size; i++) {
+        if (i < window_size / 2) {
+          k_win_[sample].data[i] = 1;
+        } else if (i == window_size / 2) {
+          k_win_[sample].data[i] = 2;
+        } else {
+          k_win_[sample].data[i] = 1;
+        }
+      }
+    }
+
+    input_.reshape(GetShape());
+    baseline_in_ = input_.cpu();
+
+    std::mt19937 rng;
+    UniformRandomFill(baseline_in_, rng, 0, 64);
+    in_ = input_.gpu();
+
+    output_.reshape(GetShape());
+    out_ = output_.gpu();
+    baseline_output_.reshape(GetShape());
+    baseline_out_ = baseline_output_.cpu();
+  }
+
+  void RunTest() {
+    KernelContext ctx_cpu, ctx_gpu;
+    KernelCpu kernel_cpu;
+    KernelGpu kernel_gpu;
+
+    auto data_shape = GetShape();
+    int num_samples = data_shape.size();
+
+    for (int sample = 0; sample < num_samples; sample++) {
+      int window_size = shape_window[sample][0];
+      auto req = kernel_cpu.Setup(ctx_cpu, data_shape[sample], window_size);
+
+      ScratchpadAllocator scratch_alloc;
+      scratch_alloc.Reserve(req.scratch_sizes);
+      auto scratchpad = scratch_alloc.GetScratchpad();
+      ctx_cpu.scratchpad = &scratchpad;
+
+      kernel_cpu.Run(ctx_cpu, baseline_out_[sample], baseline_in_[sample], k_win_[sample]);
+    }
+
+    auto req = kernel_gpu.Setup(ctx_gpu, in_.shape, shape_window);
+
+    ScratchpadAllocator scratch_alloc;
+    scratch_alloc.Reserve(req.scratch_sizes);
+    auto scratchpad = scratch_alloc.GetScratchpad();
+    ctx_gpu.scratchpad = &scratchpad;
+    kernel_gpu.Run(ctx_gpu, out_, in_, k_win_);
+
+    auto out_cpu_ = output_.cpu();
+
+    double eps = 0.01;
+    Check(out_cpu_, baseline_out_, EqualEps(eps));
+  }
+
+  TestTensorList<float, 1> kernel_window_;
+  TestTensorList<typename T::InType, T::ndim> input_;
+  TestTensorList<OutType, T::ndim> output_;
+  TestTensorList<OutType, T::ndim> baseline_output_;
+
+  TensorListView<StorageCPU, float, 1> k_win_;
+  TensorListView<StorageGPU, typename T::InType, T::ndim> in_;
+  TensorListView<StorageGPU, OutType, T::ndim> out_;
+  TensorListView<StorageCPU, typename T::InType, T::ndim> baseline_in_;
+  TensorListView<StorageCPU, OutType, T::ndim> baseline_out_;
+
+  const TensorListShape<> shape_ch_ = {{29, 145, 128, 3}, {64, 64, 64, 3},  {164, 164, 164, 3},
+                                       {12, 12, 12, 3},   {4, 200, 180, 3}, {200, 4, 180, 3},
+                                       {75, 75, 75, 5}};
+  const TensorListShape<> shape_noch_ = {{29, 145, 128}, {64, 64, 64},  {164, 164, 164},
+                                         {12, 12, 12},   {4, 200, 180}, {200, 4, 180},
+                                         {75, 75, 75}};
+  const TensorListShape<1> shape_window =
+      uniform_list_shape(shape_ch_.num_samples(), TensorShape<1>{T::window_size});
+};
+
+TYPED_TEST_SUITE_P(ConvolutionGpuKernelTest);
+
+using ConvolutionTestValues = ::testing::Types<
+    // 1D
```

Comment:
Maybe put only one case in one line and add a comment on top regarding the args. Something like:
```
//dims, has_channel, axis, window size, input type
```

Reply:
auto formatting wrapped it, will separate.